### PR TITLE
Azure Batch C# SDK version 8.1.1

### PR DESCRIPTION
**Pull request by matthchr**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by matthchr** [comment]
Note for reviewers: The "Update Batch dataplane SDK to 8.1.1" commit is the most interesting one. The "Add support for Batch OM codegenerator class remarks" commit adds a lot of files with a small change in each and is quite Batch specific so you might want to skip that one

File: src/SDKs/Batch/DataPlane/Azure.Batch/Azure.Batch.csproj [hunk 1]
```diff
@@ -10,7 +10,7 @@
     <GenerateAssemblyCompanyAttribute>false</GenerateAssemblyCompanyAttribute>
     <PackageId>Azure.Batch</PackageId>
```
Comment (shahabhijeet) [inline src/SDKs/Batch/DataPlane/Azure.Batch/Azure.Batch.csproj]: Change the package name to Microsoft.Azure.Batch

File: src/SDKs/Batch/DataPlane/Azure.Batch/Azure.Batch.csproj [hunk 2]
```diff
@@ -10,7 +10,7 @@
...
     <Description>This client library provides access to the Microsoft Azure Batch service.</Description>
-    <VersionPrefix>8.1.0</VersionPrefix>
+    <VersionPrefix>8.1.1</VersionPrefix>
```
Comment (shahabhijeet) [inline src/SDKs/Batch/DataPlane/Azure.Batch/Azure.Batch.csproj]: @matthchr add packageReleaseNotes for this release.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Batch/DataPlane/Azure.Batch/Azure.Batch.csproj (hunk 1 above)
Comment (matthchr) [reply src/SDKs/Batch/DataPlane/Azure.Batch/Azure.Batch.csproj]: I will discuss this with my team and get bck to you.

File: src/SDKs/Batch/DataPlane/Azure.Batch/Azure.Batch.csproj (hunk 2 above)
Comment (matthchr) [reply src/SDKs/Batch/DataPlane/Azure.Batch/Azure.Batch.csproj]: @shahabhijeet We already have it, it's just in a lower `<PropertyGroup>`

It says:     `<PackageReleaseNotes>For detailed release notes, see: https://aka.ms/batch-net-dataplane-changelog</PackageReleaseNotes>`

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**